pkg/clusterversion/clusterversion_test.go

```diff
@@ -1 +1,353 @@
+		name)
+
+	if apiClient == nil {
+		glog.V(100).Infof("clusterversion 'apiClient' cannot be empty")
+
+		return nil
+	}
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &corev1.ClusterVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: defaultClusterVersion,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the clusterversion is empty")
+
+		builder.errorMsg = "clusterversion 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+func TestClusterVersionPull(t *testing.T) {
+	testCases := []struct {
+		clusterVersionName  string
+		expectedError       bool
+		addToRuntimeObjects bool
+		client              bool
+		expectedErrorText   string
+	}{
+		{
+			clusterVersionName:  defaultClusterVersion,
+			expectedError:       false,
+			addToRuntimeObjects: true,
+			client:              true,
+			expectedErrorText:   "",
+		},
+		{
+			clusterVersionName:  "",
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			client:              true,
+			expectedErrorText:   "clusterversion name cannot be empty",
+		},
+		{
+			clusterVersionName:  defaultClusterVersion,
+			expectedError:       true,
+			addToRuntimeObjects: false,
+			client:              false,
+			expectedErrorText:   "clusterversion client cannot be empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, &corev1.ClusterVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testCase.clusterVersionName,
+				},
+			})
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := Pull(testSettings)
+
+		if testCase.expectedError {
+			assert.NotNil(t, err)
+
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, err.Error())
+			}
+		} else {
+			assert.Nil(t, err)
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestClusterVersionExists(t *testing.T) {
+	testCases := []struct {
+		clusterVersionExistsAlready bool
+		clusterVersionName          string
+	}{
+		{
+			clusterVersionExistsAlready: true,
+			clusterVersionName:          defaultClusterVersion,
+		},
+		{
+			clusterVersionExistsAlready: false,
+			clusterVersionName:          defaultClusterVersion,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		if testCase.clusterVersionExistsAlready {
+			runtimeObjects = append(runtimeObjects, &corev1.ClusterVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testCase.clusterVersionName,
+				},
+			})
+		}
+
+		testBuilder := NewBuilder(testSettings, testCase.clusterVersionName)
+
+		result := testBuilder.Exists()
+		if testCase.clusterVersionExistsAlready {
```

Comment:
Please remove if/else and simply use:
`assert.Equal(t, testCase.clusterVersionExistsAlready, result)`